core/mongoinit.go

```diff
@@ -0,0 +1,141 @@
+	NotificationsUser   string `json:"notifications,omitempty"`
+	NotificationsPasswd string `json:"notificationspasswd,omitempty"`
+	SchedulerUser       string `json:"scheduler,omitempty"`
+	SchedulerPasswd     string `json:"schedulerpasswd,omitempty"`
+	LoggingUser         string `json:"logging,omitempty"`
+	LoggingPasswd       string `json:"loggingpasswd,omitempty"`
+}
+
+func credentialInStore(config *tomlConfig, secretBaseURL string, credPath string, c *http.Client) (bool, error) {
+
+	t, err := getSecret(config.SecretService.TokenFolderPath + "/" + config.SecretService.VaultInitParm)
+	if err != nil {
+		lc.Error(err.Error())
+		return false, err
+	}
+
+	s := sling.New().Set(VaultToken, t.Token)
+
+	req, err := s.New().Base(secretBaseURL).Get(credPath).Request()
+	resp, err := c.Do(req)
+	if err != nil {
+		errStr := fmt.Sprintf("Failed to retrieve credentials with path as %s with error %s", credPath, err.Error())
+		return false, errors.New(errStr)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusNotFound {
+		lc.Info(fmt.Sprintf("No credential path found, setting up the credentials for %s", credPath))
+		return false, nil
+	}
+
+	lc.Info(fmt.Sprintf("%s - %d", credPath, resp.StatusCode))
+
+	var result map[string]interface{}
+
+	by, _ := ioutil.ReadAll(resp.Body)
+
+	json.Unmarshal(by, &result)
+
+	credentials := result["data"].(map[string]interface{})
+
+	if len(credentials) > 0 {
+		return true, nil
+	}
+	return false, nil
+}
+
+func initMongoDBCredentials(config *tomlConfig, secretBaseURL string, c *http.Client) error {
+
+	//TODO: we only covert coredata credential for this release, the rest needs to be implemented later.
+	adminpasswd, _ := createCredential()
+	//metadatapasswd, _ := createCredential()
+	coreadatapasswd, _ := createCredential()
+	//rulesenginepasswd, _ := createCredential()
+	//notificationspasswd, _ := createCredential()
+	//schedulerpasswd, _ := createCredential()
+	//loggingpasswd, _ := createCredential()
+
+	body := &MongoCredentials{
+		AdminUser:           "admin",
+		AdminPasswd:         adminpasswd,
+		MetadataUser:        "meta",
+		MetadataPasswd:      "password",
+		CoredataUser:        "core",
+		CoredataPasswd:      coreadatapasswd,
+		RulesengineUser:     "rules_engine_user",
+		RulesenginePasswd:   "password",
+		NotificationsUser:   "notifications",
+		NotificationsPasswd: "password",
+		SchedulerUser:       "scheduler",
+		SchedulerPasswd:     "password",
+		LoggingUser:         "logging",
+		LoggingPasswd:       "password",
+	}
+
+	t, err := getSecret(config.SecretService.TokenFolderPath + "/" + config.SecretService.VaultInitParm)
+	if err != nil {
+		lc.Error(err.Error())
+		return err
+	}
+
+	s := sling.New().Set(VaultToken, t.Token)
+
+	lc.Info("Trying to upload mongoDBinit credentials to secret service server.")
+	req, err := s.New().Base(secretBaseURL).Post(config.SecretService.MongodbinitSecretPath).BodyJSON(body).Request()
```

Comment:
This doesn't seem to match what we discussed in the WG calls a few weeks ago...

I thought that all of the secrets for core-data were to be stored in a `core-data` specific path in vault, while all the `core-metadata` secrets were to be stored in a `core-metadata` specific path in vault, etc. but here you are storing all of the secrets for all of the services inside a single secret in vault.